pyht/client.py

```diff
@@ -715,34 +716,49 @@ def _tts_ws(
         assert self._inference_coordinates is not None, "No connection"
         metrics.append("text", str(text)).append("endpoint",
                                                  str(self._inference_coordinates[voice_engine]["websocket_url"]))
-        request_id = str(uuid.uuid4())
         json_data = http_prepare_dict(text, options, voice_engine)
-        json_data["request_id"] = request_id
 
         for attempt in range(1, self._max_attempts + 1):
             try:
                 assert self._inference_coordinates is not None, "No connection"
                 ws_address = self._inference_coordinates[voice_engine]["websocket_url"]
                 if self._ws is None:
                     self._ws = connect(ws_address)
+                    self._ws_requests_sent = 0
+                    self._ws_responses_received = 0
                 try:
                     self._ws.send(json.dumps(json_data))
+                    self._ws_requests_sent += 1
                 except ConnectionClosed as e:
                     logging.debug(f"Reconnecting websocket which closed unexpectedly: {e}")
                     self._ws = connect(ws_address)
+                    self._ws_requests_sent = 0
+                    self._ws_responses_received = 0
                     self._ws.send(json.dumps(json_data))
+                    self._ws_requests_sent += 1
                 chunk_idx = -1
+                request_id = -1
+                started = False
                 for chunk in self._ws:
                     chunk_idx += 1
                     if isinstance(chunk, str):
                         msg = json.loads(chunk)
-                        if msg["type"] == "end":
+                        if msg["type"] == "start":
+                            self._ws_responses_received += 1
```

Comment:
maybe a better guarantee
```suggestion
                            self._ws_responses_received = 1
```

Reply:
See above